### DISCUSSION TITLE
Treatment Outlier

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#781] [Q&A] [answered]
Dear tim Algoritma,

Saya punya permasalahan untuk treatment data outlier untuk pemodelan logistik regression, data awal saya mayoritas outlier
<img width="498" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/472762b0-5017-4b84-b4b4-91b92e10f549">

Kemudian saya melakukan treatment dengan metode winsorizing dengan menentukan threshold 0.05 dan 0.99
syntax nya:
data_win <- winsorize(data_num %>% select(-umur_debitur), probs = c(0.05, 0.97))

kemudian saya melakukan pemilihan variabel dengan melihat korelasi data yang nilainya > 0.6 yang saya ambil.

correlation_matrix <- cor(data_win %>% 
                            select(-baki_debet))
threshold <- 0.6
correlated_variables <- which(abs(correlation_matrix) > threshold & correlation_matrix != 1, arr.ind = TRUE)

selected_variables <- colnames(data_win %>% 
                                 select(-baki_debet))[unique(c(correlated_variables[, 1], correlated_variables[, 2]))]
df <- data_win[, selected_variables]
selected_variables

Namun setelah saya melakukan treatment winsorizing pun masih ada data outlier dari variabel yang saya perlukan untuk pemodelan. Hasilnya seperti berikut:
<img width="484" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/5b2653fa-8b3d-4a12-a0ce-26c2d5c614d2">

Saya memilih untuk menggunakan metode winsorizing dengan alasan, observasi data yang hanya sedikit sehingga saya ingin mempertahankan data outlier dan variabel-variabel yang saya pertahankan merupakan variabel yang penting (dalam konteks kepentingan dari aktivitas bisnis). Apakah ada metode lain yang bisa digunakan selain membuang outlier atau menghilangkan variabel.

Juga sebaiknya gimana ya, apakah melakukan treatment outlier terlebih dahulu kemudian seleksi variabel atau sebaliknya. Karena kalo saya lihat dari korelasinya setelah melakukan winsorizing dengan sebelum nilainya tidak berbeda jauh.

Mohon petunjuknya,

Best regards,
JT

**Comment by saskia-dwi-ulfah** [comment]
Dear @joejnt , terima kasih atas pertanyaannya.

1. Untuk handling outlier, salah satu pendekatan yang bisa dieksplorasi adalah melakukan transformasi, seperti menggunakan tranformasi `log` pada prediktor. Yang harus diperhatikan bahwa saat melakukan transformasi, berarti nilai data kita adalah hasil transformasi, bukan lagi data asli.
2. Jika melihat dari gambar yang dilampirkan, sepertinya nature dari datanya memang kebanyakan outlier. Bisa dipertimbangkan untuk menggunakan metode yang lebih robust selain logistic regression, misalnya decision tree, random forest, atau gradient boosting. Berikut referensi yang bisa coba didalami:
https://www.javatpoint.com/r-classification
https://www.geeksforgeeks.org/classification-in-r-programming/
https://www.geeksforgeeks.org/decision-tree-in-r-programming/
https://www.geeksforgeeks.org/random-forest-approach-in-r-programming/
https://datascienceplus.com/gradient-boosting-in-r/
3. Untuk apakah handling outlier dulu atau seleksi variabel, biasanya handling outlier lebih dahulu dilakukan.


Semoga jawaban ini dapat membantu.

Best regards,
Team algoritma

**Comment by joejnt** [comment]
Dear @saskia-dwi-ulfah 

Terima kasih untuk informasinya, untuk penggunaan transformasi log() apakah bisa di share referensi dokumentasinya.

Best regards,
JT

**Comment by saskia-dwi-ulfah** [comment, marked as answer]
Dear @joejnt 

Berikut referensi terkait implementasi log pada R
1.[ How to do log transformation in R-Complete Guide](https://finnstats.com/2022/06/27/how-to-do-log-transformation-in-r/)
2. [log: Logarithms and Exponentials](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/log)

Semoga jawaban ini dapat membantu.

Best regards, 
Team algoritma

**Comment by joejnt** [comment]
Dear @saskia-dwi-ulfah 

Terima kasih untuk informasinya.

Best regards,
JT
